x-pack/plugins/cloud_security_posture/common/utils/helpers.ts

```diff
@@ -98,3 +100,29 @@ export const getStatusForIndexName = (indexName: string, status?: BaseCspSetupSt
 
   return 'unknown';
 };
+
+export const cleanupCredentials = (packagePolicy: NewPackagePolicy | UpdatePackagePolicy) => {
```

Comment:
The function does its job well, here are some subjective comments on the readability and safety of the function:
- we seem to rely on the ` enabledInput?.streams?.[0].vars`, what about returning early if it's not there? This can help avoid the check when you remove unwanted credentials. Plus we rely on the fact that there is exactly one element in `streams`, I don't know how safe it is
- The use of `reduce` is a bit confusing because we are not reducing to a single value, but using it more instead of `forEach` or `for..in`
- We mutate the `packagePolicy` directly, might be a bit safer to avoid the mutation and work with a copy of a package policy